clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -245,6 +245,9 @@ def err_invalid_vector_long_double_decl_spec : Error<
 def warn_vector_long_decl_spec_combination : Warning<
   "Use of 'long' with '__vector' is deprecated">, InGroup<Deprecated>;
 
+def err_static_decl_uses_free_type_variable : Error<
```

Comment:
I think the error messages needs to be changed to be more understandable by C programmers.  Most of them won't know what a free type variable.   How about something like the following?  "static variable cannot have a type variable bound by an enclosing scope"

Reply:
I think you need to include the type and indicate in the message that the problem is with type.  Something akin to:

"type for static variable '%0' cannot use a type variable '%1' that is bound by an enclosing scope"

or 

"static variable '%0' has a type '%1' that uses a type variable bound by an enclosing scope'